Sources/MapboxCoreNavigation/Location/RoadObjectLocation.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import MapboxNavigationNative
+import Turf
+
+/**
+ The location of a road object in the road graph.
+ */
+public enum RoadObjectLocation {
+
+    /** Location of an object represented as a gantry */
+    case gantry(GantryLocation)
+
+    /** Location of an object represented as a point */
+    case point(PointLocation)
+
+    /** Location of an object represented as a polygon */
+    case polygon(PolygonLocation)
+
+    /** Location of an object represented as a polyline */
+    case polyline(PolylineLocation)
+
+    /** Location of an object represented as an OpenLR line */
+    case openLRLine(OpenLRLineLocation)
+
+    /** Location of an object represented as an OpenLR point */
+    case openLRPoint(OpenLRPointLocation)
+
+    /** Location of a route alert */
+    case routeAlert(RouteAlertLocation)
```

Comment:
For simplicity, inline the various “Location” types as individual associated values, since they’re only used here.